### PR TITLE
Github workfow + run_tests.sh option to run marked integration tests on legacy apis

### PR DESCRIPTION
**Pull request by jdavcs**
Addresses #11321. Does this work as a solution?

### CODE REVIEW

**Comment by mvdbeek** [comment]
I guess it would, but this seems like unnecessary noise and overhead if we run that on each PR.
Can you change that to a cron job that runs at night?

**Comment by ic4f** [comment]
> I guess it would, but this seems like unnecessary noise and overhead if we run that on each PR.
> Can you change that to a cron job that runs at night?

Yes - done, but commented out. First, I want to check that wrapping the run command works (need to change exit status from 5 to 0 if no tests are run). - and it seems it's not working..

**Comment by ic4f** [comment]
So the new legacy tests fail when they don't find any tests to run - which is the case on this first PR (which is expected); but this will be also the case anytime there are no tests marked for execution. The reason for that is that pytest will exit with status 5 if no tests were run (which is the correct behavior). I don't see an obvious way to capture the exit status of the command within the workflow yaml file. I don't have a good solution yet. It's possible, of course, to capture the exit status inside run_tests.sh (and exit with 0 if it was 5) - but that would require modifying `run_tests.sh`. If that is the only option, I could add a flag to run_tests.sh that does that. 
But I'm not sure if this is a good solution..

File: .github/workflows/integration_legacy.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        if: matrix.subset == 'kubernetes'
+        run: |
+          . .ci/minikube-test-setup/start_services.sh
+          ./run_tests.sh -integration test/integration --legacy-api -- -k "${{ matrix.subset }}"
+        working-directory: 'galaxy root'
+      - uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: Integration test results (${{ matrix.python-version }}, ${{ matrix.subset }})
+          path: 'galaxy root/run_integration_tests.html'
```
Comment (mvdbeek) [inline .github/workflows/integration_legacy.yaml:79]: You can collapse this into one, that should solve the issue of no tests in these chunks.
We only split them to reduce runtime, which is not a concern here.

**Comment by mvdbeek** [comment]
We're working towards removing the legacy API endpoints ASAP, ideally shortly after the 21.05 release, so maybe it's not worth pursuing this further ? I'll drop the milestone for now

**Comment by ic4f** [comment]
> We're working towards removing the legacy API endpoints ASAP, ideally shortly after the 21.05 release, so maybe it's not worth pursuing this further ? I'll drop the milestone for now

I was just thinking about this! So yes, +1, closing it.

**Pull request closed without merging**